tests/framework/db/ActiveRecordTest.php

```diff
@@ -527,6 +527,7 @@ public function testInverseOf()
         $this->assertTrue($customer->orders2[0]->customer2 === $customer);
         $this->assertTrue($customer->orders2[1]->customer2 === $customer);
         // ad-hoc lazy loading
+        // NB: this batch of asserts probably does not do what was intended ($customer->orders2[0] vs $orders[0])
```

Comment:
good catch, if this test would be written correctly, it would have already cought the bug. :)